### PR TITLE
New if_switch_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #884

No hits on {lintr}

### CODE REVIEW

File: inst/lintr/linters.csv [hunk 1]
```diff
@@ -36,6 +36,7 @@ function_argument_linter,style consistency best_practices
 function_left_parentheses_linter,style readability default
 function_return_linter,readability best_practices
 if_not_else_linter,readability consistency configurable
+if_switch_linter,best_practices readability consistency
```
Comment (AshesITR) [inline inst/lintr/linters.csv:39]: The docs suggest efficiency.

**Review by AshesITR — COMMENTED** [review]
Missing metadata and vectorization tests.

File: R/if_switch_linter.R [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "if (x == 'a') 1 else if (x == 'b') 2 else 3",
+#'   linters = if_switch_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "if (x == 'a') 1 else if (x == 'b' & y == 2) 2 else 3",
```
Comment (IndrajeetPatil) [inline R/if_switch_linter.R:26]: Shouldn't this example use `switch()`?

Comment (MichaelChirico) [reply R/if_switch_linter.R:26]: note the condition on y

Comment (IndrajeetPatil) [reply R/if_switch_linter.R:26]: No, I saw that, but it seems odd to provide "okay" example that doesn't use `switch()` in examples for a linter that vouches for using `switch()`. 

Maybe we can include both variants of these okay examples?

File: R/if_switch_linter.R [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        "Prefer switch() statements over repeated if/else equality tests,",
+        "e.g., switch(x, a = 1, b = 2) over",
+        'if (x == "a") 1 else if (x == "b") 2.'
+      ),
+      type = "warning"
+    )
+  })
+}
+
+# TODO(michaelchirico): set the lint to apply whenever there are <=3 child
```
Comment (AshesITR) [inline R/if_switch_linter.R:86]: What's with theses todos?

Comment (AshesITR) [reply R/if_switch_linter.R]: Okay do you want to keep track of them in both places or remove the todos from the code?

**Review by AshesITR — APPROVED** [review]

**Pull request merged**